src/parse/parser.rs

```diff
@@ -1280,6 +1280,17 @@ where
             )?);
         } else {
             debug!("None");
+            if needs_eq && min_vals_zero {
+                // OPTION with missing optional value?
+                if !opt.default_missing_vals.is_empty() {
+                    // debug!("Parser::add_value:iter:{}: has default vals", arg.name);
+                    debug!("Has default_missing_vals");
+                    for val in &opt.default_missing_vals {
+                        debug!("Adding value from default_missing_values; val = {:?}", val);
+                        self.add_val_to_arg(opt, &ArgStr::new(val), matcher, ValueType::CommandLine)?; // ? ValueType::DefaultValue
```

Comment:
I think `CommandLine` is correct here. @CreepySkeleton What do you think?

Reply:
TL;DR Yes, CommandLine is correct

This is a good example of the distinction between the option itself n its values. 
```
   --foo val1
   ----- ----
 present present

   --foo
   ----- ----
 present absent: default
```

So the _option_ is indeed present on the command line and should be marked as `CommandLine`, but _value_ is absent so it _should_ have been marked as `DefaultValue`, but we don't mark values, only options. It doesn't change anything as far as I see, but we may want to annotate values as well in future if something happen to justify if.